Library/Homebrew/formula.rb

```diff
@@ -1013,9 +1018,10 @@ def _outdated_versions(options = {})
 
     installed_kegs.each do |keg|
       version = keg.version
+      tab = Tab.for_keg(keg)
       all_versions << version
 
-      return [] if pkg_version <= version && !version.head?
+      return [] if pkg_version <= version && epoch == tab.epoch && !version.head?
```

Comment:
Can you split this into multiple statements/lines somehow? Quite a bit of logic being chained here; perhaps having 2-3 separate variables would help.